generic/ltp.py

```diff
@@ -65,22 +67,17 @@ def test(self):
         ltpbin_dir = os.path.join(self.workdir, "ltp-master", 'bin')
         cmd = os.path.join(ltpbin_dir, 'runltp') + ' ' + args
         result = process.run(cmd, ignore_status=True)
-        # Walk the stdout and try detect failed tests from lines like these:
-        # aio01       5  TPASS  :  Test 5: 10 reads and writes in  0.000022 sec
-        # vhangup02    1  TFAIL  :  vhangup02.c:88: vhangup() failed, errno:1
-        # and check for fail_statuses The first part contain test name
-        fail_statuses = ['TFAIL', 'TBROK', 'TWARN']
-        split_lines = (line.split(None, 3)
-                       for line in result.stdout.splitlines())
-        failed_tests = [items[0] for items in split_lines
-                        if len(items) == 4 and
-                        items[2].strip(":") in fail_statuses]
+        # Walk the ltp.log and try detect failed tests from lines like these:
+        # msgctl04                                           FAIL       2
+        with open(logfile, 'r') as fp:
+            lines = fp.readlines()
+            for line in lines:
+                if 'FAIL' in line:
+                    value = re.split(r'\s+', line)
+                    self.failed_tests.append(value[0])
 
-        if failed_tests:
-            self.fail("LTP tests failed: %s" % ", ".join(failed_tests))
-        elif result.exit_status != 0:
-            self.fail("No test failures detected, but LTP finished with %s"
-                      % (result.exit_status))
+        if len(self.failed_tests) != 0:
```

Comment:
Minor suggestion: `if self.failed_tests:` would suffice here and also abides pylint standard.

Reply:
I think both forms are valid .. but yeah this can be changed to just self.failed_tests.  Will send updated patch